pkg/gen/filters/common/helper.go

```diff
@@ -45,3 +47,24 @@ func Pluralize(s string, i int) string {
 	}
 	return plural.Plural(s)
 }
+
+func MakeListOfFields[T any](inputList []T, fieldName string) ([]string, error) {
+	list := []string {}
+	for _, element := range inputList {
+	    r := reflect.ValueOf(element)
+		reflectValue := reflect.Indirect(r).FieldByName(fieldName)
+		if !reflectValue.IsValid() {
+			return list, fmt.Errorf("given struct %T has no field %s ",  element, fieldName)
+		}
+		value := reflectValue.String()
+		list = append(list, value)
+	}
+	return list, nil
+}
+
+func RemoveDuplicates (inputList []string) []string {
```

Comment:
Unique might be a better name for this. And it could take a constraints.Ordered as type, as this is expected by sort and shoudl also work with compact.
Not sure why you assign inputList to list, the slice will not be copied. It will modify the original slice. return slices.Compact(slices.Sort(inputList)).
To make a copy use slices.Clone.
In general slices are pointers to a list passed as values. Like QList (ref counting).